subworkflows/nf-core/bam_downsample_samtools/main.nf

```diff
@@ -0,0 +1,54 @@
+include { SAMTOOLS_DEPTH } from '../../../modules/nf-core/samtools/depth'
+include { GAWK           } from '../../../modules/nf-core/gawk'
+include { SAMTOOLS_VIEW  } from '../../../modules/nf-core/samtools/view'
+
+workflow BAM_DOWNSAMPLE_SAMTOOLS {
+
+    take:
+    ch_bam    // channel: [ [id], bam, bai ]
+    ch_fasta  // channel: [ [genome], fasta ]
+    depth     // channel: [ depth ]
```

Comment:
If you want this subworkflow to be able to downsample different samples to different depths (looks like it in the tests?), then I would input the depth _together_ with the bam files, e.g:

```suggestion
    ch_bam_bai_depth    // channel: [ [id], bam, bai, depth ]
    ch_fasta  // channel: [ [genome], fasta ]
```

If you want to downsample all samples to the same depth, I think it's fine as it is.

Reply:
I agree, it would make more sense.
I'll update everything.